server/sheet.go

```diff
@@ -69,7 +69,164 @@ func (s *Server) registerSheetRoutes(g *echo.Group) {
+		}
+
+		sheetFind := &api.SheetFind{
+			ID: &id,
+		}
+		sheet, err := s.store.GetSheet(ctx, sheetFind, currentPrincipalID)
+		if err != nil {
+			return echo.NewHTTPError(http.StatusInternalServerError, fmt.Sprintf("Failed to fetch sheet ID: %v", id)).SetInternal(err)
+		}
+		if sheet == nil {
+			return echo.NewHTTPError(http.StatusNotFound, fmt.Sprintf("sheet ID not found: %d", id))
+		}
+
+		c.Response().Header().Set(echo.HeaderContentType, echo.MIMEApplicationJSONCharsetUTF8)
+		if err := jsonapi.MarshalPayload(c.Response().Writer, sheet); err != nil {
+			return echo.NewHTTPError(http.StatusInternalServerError, fmt.Sprintf("Failed to marshal sheet response: %v", id)).SetInternal(err)
+		}
+		return nil
+	})
+
+	g.PATCH("/sheet/:sheetID", func(c echo.Context) error {
+		ctx := c.Request().Context()
+		currentPrincipalID := c.Get(getPrincipalIDContextKey()).(int)
+		id, err := strconv.Atoi(c.Param("sheetID"))
+		if err != nil {
+			return echo.NewHTTPError(http.StatusBadRequest, fmt.Sprintf("ID is not a number: %s", c.Param("sheetID"))).SetInternal(err)
+		}
+
+		sheetPatch := &api.SheetPatch{
+			ID:        id,
+			UpdaterID: currentPrincipalID,
+		}
+		if err := jsonapi.UnmarshalPayload(c.Request().Body, sheetPatch); err != nil {
+			return echo.NewHTTPError(http.StatusBadRequest, "Malformed patch sheet request").SetInternal(err)
+		}
+
+		sheet, err := s.store.PatchSheet(ctx, sheetPatch)
+		if err != nil {
+			if common.ErrorCode(err) == common.NotFound {
+				return echo.NewHTTPError(http.StatusNotFound, fmt.Sprintf("sheet ID not found: %d", id))
+			}
+			return echo.NewHTTPError(http.StatusInternalServerError, fmt.Sprintf("Failed to patch sheet with ID: %d", id)).SetInternal(err)
+		}
+
+		c.Response().Header().Set(echo.HeaderContentType, echo.MIMEApplicationJSONCharsetUTF8)
+		if err := jsonapi.MarshalPayload(c.Response().Writer, sheet); err != nil {
+			return echo.NewHTTPError(http.StatusInternalServerError, fmt.Sprintf("Failed to marshal patch sheet response: %d", id)).SetInternal(err)
+		}
+		return nil
+	})
+
+	g.DELETE("/sheet/:sheetID", func(c echo.Context) error {
+		ctx := c.Request().Context()
+		id, err := strconv.Atoi(c.Param("sheetID"))
+		if err != nil {
+			return echo.NewHTTPError(http.StatusBadRequest, fmt.Sprintf("ID is not a number: %s", c.Param("sheetID"))).SetInternal(err)
+		}
+
+		sheetDelete := &api.SheetDelete{
+			ID:        id,
+			DeleterID: c.Get(getPrincipalIDContextKey()).(int),
+		}
+		err = s.store.DeleteSheet(ctx, sheetDelete)
+		if err != nil {
+			return echo.NewHTTPError(http.StatusInternalServerError, fmt.Sprintf("Failed to delete sheet ID: %v", id)).SetInternal(err)
+		}
+
+		c.Response().Header().Set(echo.HeaderContentType, echo.MIMEApplicationJSONCharsetUTF8)
+		c.Response().WriteHeader(http.StatusOK)
+		return nil
+	})
+
+	g.POST("/project/:projectID/sync-sheet", func(c echo.Context) error {
```

Comment:
Since this is moved under /project, let's move it to the project.go handler.

Also, let's try to avoid other refactoring if possible.